File: hacking_with_python/debug_endpoints.py
import requests
from bs4 import BeautifulSoup
from urllib.parse import urljoin
from termcolor import colored

def find_debug_endpoints(url):
    print(colored("[*] Discovering potential debug endpoints on:", "cyan"), colored(url, "blue"))
    debug_endpoints = []

    try:
        response = requests.get(url)
        if response.status_code == 200:
            soup = BeautifulSoup(response.text, "html.parser")
            links = soup.find_all("a", href=True)
            for link in links:
                href = link["href"]
                if any(keyword in href.lower() for keyword in ["debug", "test", "admin", "dev", "trace", "info", "logs", "log", "debugger", "monitor", "diagnose", "inspect", "console", "audit", "status", "check", "health", "metrics"]):
                    debug_endpoints.append(href)

            return debug_endpoints

    except Exception as e:
        print(colored("[!] Error:", "red"), e)

def test_vulnerability(debug_endpoints):
    if debug_endpoints:
        print(colored("[+] Potential debug endpoints found:", "green"))
        for endpoint in debug_endpoints:
            print(colored("[+] ", "green"), endpoint)
        print(colored("[*] The website may be vulnerable.", "yellow"))
        print(colored("[*] To test the vulnerability, try accessing these endpoints and observe the response.", "yellow"))
    else:
        print(colored("[*] No potential debug endpoints found.", "green"))
        print(colored("[*] The website seems not to be vulnerable.", "yellow"))

def main():
    target_website = input(colored("Enter your target website url: ", "cyan"))
    debug_endpoints = find_debug_endpoints(target_website)
    test_vulnerability(debug_endpoints)

if __name__ == "__main__":
    main()

File: hacking_with_ruby/debug_endpoints.rb
require 'net/http'
require 'uri'
require 'colorize'

def test_debug_endpoints(url)
  puts "Testing for Exposed Debug Endpoints on #{url}...".colorize(:yellow)
  puts "-----------------------------".colorize(:yellow)
  
  debug_endpoints = [
    'debug', 'debug.php', 'debug.aspx', 'debug.jsp', 'debug.py', 'debug.rb', 'debug.pl',
    'test', 'test.php', 'test.aspx', 'test.jsp', 'test.py', 'test.rb', 'test.pl',
    'dev', 'dev.php', 'dev.aspx', 'dev.jsp', 'dev.py', 'dev.rb', 'dev.pl',
    'logs', 'logs.php', 'logs.aspx', 'logs.jsp', 'logs.py', 'logs.rb', 'logs.pl',
    'admin', 'admin.php', 'admin.aspx', 'admin.jsp', 'admin.py', 'admin.rb', 'admin.pl',
    'info', 'info.php', 'info.aspx', 'info.jsp', 'info.py', 'info.rb', 'info.pl',
    'status', 'status.php', 'status.aspx', 'status.jsp', 'status.py', 'status.rb', 'status.pl'
  ]
  
  vulnerable_endpoints = []
  
  debug_endpoints.each do |endpoint|
    endpoint_url = "#{url}/#{endpoint}"
    puts "Checking #{endpoint_url}...".colorize(:light_blue)
    
    uri = URI.parse(endpoint_url)
    response = Net::HTTP.get_response(uri)
    
    if response.code == "200"
      vulnerable_endpoints << endpoint_url
      puts "#{endpoint_url} is vulnerable!".colorize(:red)
    else
      puts "#{endpoint_url} is not vulnerable.".colorize(:green)
    end
  end
  
  puts "-----------------------------".colorize(:yellow)
  
  if vulnerable_endpoints.empty?
    puts "No Exposed Debug Endpoints found on #{url}.".colorize(:green)
  else
    puts "Vulnerable endpoints found:".colorize(:red)
    vulnerable_endpoints.each { |endpoint| puts "  #{endpoint}" }
  end
end

def test_vulnerability(url)
  test_debug_endpoints(url)
end

def main()
  puts "Enter your target website url: ".colorize(:yellow)
  target_website = gets.chomp.strip
  
  test_vulnerability(target_website)
end

main()
